test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -7276,6 +7276,146 @@ public void PackCommand_RequireLicenseAcceptanceNotEmitted()
+                    workingDirectory,
+                    "pack Package.nuspec",
+                    waitForExit: true);
+                Assert.True(0 == specResult.ExitCode, specResult.AllOutput);
+                Assert.True(0 == packResult.ExitCode, packResult.AllOutput);
+
+                // Assert
+                var path = Path.Combine(workingDirectory, "Package.1.0.0.nupkg");
+                using (var package = new PackageArchiveReader(path))
+                {
+                    var files = package.GetNonPackageDefiningFiles().OrderBy(s => s).ToArray();
+
+                    Assert.Equal(
+                        new string[]
+                        {
+                        "data.txt",
+                         "images/1.png",
+                         "lib/uap10.0/a.dll",
+                         "tools/install.ps1",
+                        },
+                        files);
+
+                    Assert.False(packResult.Item2.Contains("Assembly outside lib folder"));
+                }
+            }
+        }
+
+        [Fact]
+        public void PackCommand_WithRepo_WithDefaultNuspec_Succeeds()
+        {
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var workingDirectory = TestDirectory.Create())
+            {
+                // Arrange
+                var projectName = Path.GetFileName(workingDirectory);
+
+                Util.CreateFile(
+                    workingDirectory,
+                    "proj1.csproj",
+    @"<Project ToolsVersion='4.0' DefaultTargets='Build'
+    xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <OutputPath>out</OutputPath>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include='file1.cs' />
+  </ItemGroup>
+  <Import Project='$(MSBuildToolsPath)\Microsoft.CSharp.targets' />
+</Project>");
+
+                Util.CreateFile(
+                    workingDirectory,
+                    "file1.cs",
+    @"using System;
+using System.Reflection;
+
+[assembly: AssemblyVersion(" + "\"1.0.0.0\"" + @")]
+namespace proj1
+{
+    public class Class1
+    {
+        public int A { get; set; }
+    }
+}");
+                Util.CreateFile(
+                    workingDirectory,
+                    "data.txt",
+                    string.Empty);
+
+                // Act
+                CommandRunnerResult specResult = CommandRunner.Run(
+                    nugetexe,
+                    workingDirectory,
+                    "spec",
+                    waitForExit: true);
+                CommandRunnerResult packResult = CommandRunner.Run(
+                    nugetexe,
+                    workingDirectory,
+                    " pack -properties tagVar=CustomTag;author=microsoft.com;Description=aaaaaaa -build",
+                    waitForExit: true);
+                Assert.True(0 == specResult.ExitCode, specResult.AllOutput);
```

Comment:
It's probably worthwhile moving the `specResult.ExitCode` assert to before `packResult`'s `CommandRunner.Run`, so we don't waste time running a command when the pre-requisites are failed.